preprocessing/structs.ts

```diff
@@ -160,7 +160,7 @@ library OfferUnpackedExtra {
   }
   function bin(OfferUnpacked memory offer, uint tickSpacing) internal pure returns (Bin) {
     // Offers are always stored with a tick that corresponds exactly to a tick
-    return BinLib.fromBinAlignedTick(offer.tick, tickSpacing);
+    return offer.tick.alignedToNearestBin(tickSpacing);
```

Comment:
This reads a bit weird. Not sure how to improve it though

Reply:
In 6232ec08cfa33cb5f609747d6962a9f289e3d775 I replace both tick->bin functions with a single optimized assembly function. The move to assembly negates any optimization and in gas tests reduces costs by ~140 gas most of the time, up to ~1000 gas (no gas increase observed).